lightning/src/ln/channel_keys.rs

```diff
@@ -0,0 +1,249 @@
+use crate::io;
+use crate::util::ser::Writer;
+use crate::util::ser::Writeable;
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::hashes::sha256::Hash as Sha256;
+
+macro_rules! doc_comment {
+    ($x:expr, $($tt:tt)*) => {
+        #[doc = $x]
+        $($tt)*
+    };
+}
+macro_rules! basepoint_impl {
+    ($BasepointT:ty) => {
+        impl $BasepointT {
+            /// Get inner Public Key
+            pub fn to_public_key(&self) -> PublicKey {
+                self.0
+            }
+        }
+        
+        impl From<PublicKey> for $BasepointT {
+            fn from(value: PublicKey) -> Self {
+                Self(value)
+            }
+        }
+        
+    }
+}
+macro_rules! key_impl {
+    ($BasepointT:ty, $KeyName:expr) => {
+        doc_comment! {
+            concat!("Generate ", $KeyName, " using per_commitment_point"),
+            pub fn from_basepoint<T: secp256k1::Signing>(
+                secp_ctx: &Secp256k1<T>,
+                basepoint: &$BasepointT,
+                per_commitment_point: &PublicKey,
+            ) -> Self {
+                Self(derive_public_key(secp_ctx, per_commitment_point, &basepoint.0))
+            }
+        }
+        
+        doc_comment! {
+            concat!("Generate ", $KeyName, " from privkey"),
+            pub fn from_secret_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, sk: &SecretKey) -> Self {
+                Self(PublicKey::from_secret_key(&secp_ctx, &sk))
+            }
+        }
+        
+        /// Get inner Public Key
+        pub fn to_public_key(&self) -> PublicKey {
+            self.0
+        }
+    }
+}
+macro_rules! key_read_write {
+    ($SelfT:ty) => {
+        impl Writeable for $SelfT {
+            fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+                self.0.serialize().write(w)
+            }
+        
+            fn serialized_length(&self) -> usize {
+                PUBLIC_KEY_SIZE
+            }
+        }
+        
+        impl Readable for $SelfT {
+            fn read<R: io::Read>(r: &mut R) -> Result<Self, DecodeError> {
+                let key: PublicKey = Readable::read(r)?;
+                Ok(Self(key))
+            }
+        }
+    }
+}
+
+
+
+/// Used to generate [`local_delayedpubkey`](https://github.com/lightning/bolts/blob/master/03-transactions.md#key-derivation) for the latest state of a channel.
+#[derive(PartialEq, Eq, Clone, Copy, Debug, Hash)]
+pub struct DelayedPaymentBasepoint(PublicKey);
+basepoint_impl!(DelayedPaymentBasepoint);
+key_read_write!(DelayedPaymentBasepoint);
+
+/// [delayedpubkey](https://github.com/lightning/bolts/blob/master/03-transactions.md#localpubkey-local_htlcpubkey-remote_htlcpubkey-local_delayedpubkey-and-remote_delayedpubkey-derivation)
+/// Used to regenerate the scripts required for a penalty transaction or a spend from a force closed channel.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub struct DelayedPaymentKey(PublicKey);
```

Comment:
I personally enjoy that there is a an explicit step to put something into `DelayedPaymentKey`, but happy to remove those frictions as it may annoy more advanced users. `to_public_key()` is a helper, so one don't have to do `payment_key.0`, which is a bit cryptic.